From 03c9a40883e610e54b8fc3c3ce1842d68d7ce2e4 Mon Sep 17 00:00:00 2001
From: Kefu Chai <tchaikov@gmail.com>
Date: Mon, 15 Jul 2024 09:27:16 +0800
Subject: [PATCH] blocks,runtime: io_signature: include spdlog/*/ranges.h for
 using fmt::join()

fmt::join() was moved into fmt/ranges.h since fmt 11, so let's
include the corresponding header in spdlog for using it.

Signed-off-by: Kefu Chai <tchaikov@gmail.com>
(cherry picked from commit 19b070051c1c2b5fb6f2da8fb6422b27418c3dfa)
Signed-off-by: Jeff Long <willcode4@gmail.com>
Upstream: https://github.com/gnuradio/gnuradio/commit/ead459813367e7fd679dad067d1b020010d49b4f
[Julien: backported commit ead45981 from v3.10.11.0]
Signed-off-by: Julien Olivain <ju.o@free.fr>
---
 gr-blocks/lib/message_debug_impl.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gr-blocks/lib/message_debug_impl.cc b/gr-blocks/lib/message_debug_impl.cc
index 1e81aec0d..415d2be97 100644
--- a/gr-blocks/lib/message_debug_impl.cc
+++ b/gr-blocks/lib/message_debug_impl.cc
@@ -14,6 +14,11 @@
 #include <pmt/pmt.h>
 #include <spdlog/common.h>
 #include <spdlog/fmt/fmt.h>
+#if __has_include(<spdlog/fmt/ranges.h>)
+#include <spdlog/fmt/ranges.h>
+#elif __has_include(<spdlog/fmt/bundled/ranges.h>)
+#include <spdlog/fmt/bundled/ranges.h>
+#endif
 #include <functional>
 #include <utility>
 #include <vector>
-- 
2.46.0

